src/lib.rs

```diff
@@ -286,17 +263,101 @@ _start:
     lui ra, %hi(_abs_start)
     jr %lo(_abs_start)(ra)
 
-// .section .text
-
 _abs_start:
     .option norelax
     .cfi_startproc
     .cfi_undefined ra
-
-    // Unsupported on ESP32-C2/C3
-    // csrw mie, 0
-    // csrw mip, 0
-
+"#
+);
+#[cfg(feature = "has-mie-mip")]
```

Comment:
Not sure how LLVM/Rust handles the ordering multiple `global_asm!` statements, we should merge this to a single statement using https://github.com/rust-embedded/cortex-m/blob/067af8406f306bf8fba53ce738e441a8ba7fefe1/cortex-m-rt/src/lib.rs#L485-L501

Reply:
Thanks for the pointer! I was also concerned about it but for some reason I didn't realized I could just write a macro 😄